### DISCUSSION TITLE
input focus was delayed when submit

### DISCUSSION THREAD

**Discussion by dhieunguyen** [react-hook-form/react-hook-form#5352] [Q&A] [answered]
**Describe the bug**
Can you guys help me in this case. When i submitted my form, input was focus but it delayed a little time. Please watch the video. I'm using v7. Thanks for reading

Uploading 3796217609563621237.mp4…

**Comment by dhieunguyen** [comment]
https://user-images.githubusercontent.com/64343185/119765478-89918f80-bedd-11eb-8a95-3002b6d769ae.mp4

**Comment by bluebill1049** [comment]
Do you have any Async validation?

**Comment by dhieunguyen** [comment, marked as answer]
can u explain it or give me an example ? Thanks

**Reply by bluebill1049** [reply]
are you doing any validation async? 

eg

`validate: async () => return fetch();`

**Reply by dhieunguyen** [reply]
hmmm i don't even call any api request

**Reply by bluebill1049** [reply]
I have no idea then unless you can reproduce that in a codesandbox, it should focus right after validation.

**Reply by dhieunguyen** [reply]
thanks for your supports, i think the problem was from me and i will try to fix it <3

**Reply by bluebill1049** [reply]
👍

**Reply by dhieunguyen** [reply]
@bluebill1049 sorry for my question. I think my form was delayed to focus because it's was re-render when i clicked submit, my form has 9 inputs and another component. Can i avoid it?
